src/Core/Environment.php

```diff
@@ -151,4 +166,93 @@ public static function getTimeLimitMax()
     {
         return static::$timeLimitMax;
     }
+
+    /**
+     * Get value of environment variable
+     *
+     * @param string $name
+     * @return mixed Value of the environment variable, or null if not set
+     */
+    public static function getEnv($name)
+    {
+        switch (true) {
+            case array_key_exists($name, static::$env):
+                return static::$env[$name];
+            case array_key_exists($name, $_ENV):
+                return $_ENV[$name];
+            case array_key_exists($name, $_SERVER):
+                return $_SERVER[$name];
+            default:
+                return getenv($name);
+        }
+    }
+
+    /**
+     * Set environment variable using php.ini syntax.
+     * Acts as a process-isolated version of putenv()
+     * Note: This will be parsed via parse_ini_string() which handles quoted values
+     *
+     * @param string $string Setting to assign in KEY=VALUE or KEY="VALUE" syntax
+     */
+    public static function putEnv($string)
+    {
+        // Parse name-value pairs
+        $envVars = parse_ini_string($string) ?: [];
+        foreach ($envVars as $name => $value) {
+            self::setEnv($name, $value);
+        }
+    }
+
+    /**
+     * Load environment variables from .env file
+     *
+     * @param string $path Path to the file
+     * @param bool $overload Set to true to allow vars to overload. Recommended to leave false.
+     * @return array List of values parsed as an associative array
+     */
+    public static function putEnvFromFile($path, $overload = false)
+    {
+        // Parse and cleanup content
+        $contents = implode(
+            PHP_EOL,
+            array_filter(array_map(
+                function ($line) {
+                    // Strip all # comments, including start or end of string, but not in quoted strings
+                    // '#comment', 'var=thing#comment', or 'var="not#comment"#comment
+                    return preg_replace('/((^[^"]*)|(^[^"]+("[^"]*){2}))#.*$/', '$1', $line);
+                },
+                explode(PHP_EOL, file_get_contents($path))
+            ))
+        );
+
+        // Pass to parse_ini_string
+        $variables = parse_ini_string($contents) ?: [];
+        foreach ($variables as $name => $value) {
+            // Conditionally prevent overloading
+            if ($overload || static::getEnv($name) === false) {
+                static::setEnv($name, $value);
+            }
+        }
+        return $variables;
+    }
+
+    /**
+     * Set environment variable via $name / $value pair
+     *
+     * @param string $name
+     * @param string $value
+     */
+    public static function setEnv($name, $value)
+    {
+        // If PHP is running as an Apache module and an existing
+        // Apache environment variable exists, overwrite it
+        if (function_exists('apache_getenv') && function_exists('apache_setenv') && apache_getenv($name)) {
+            apache_setenv($name, $value);
```

Comment:
I don't understand the preference for handrolling our own dotenv work when this is practically identical to using `Dotenv\Loader::setEnvironmentVariable` and `Dotenv\Loader::getEnvironmentVariable` and means we don't have to maintain this code :/

Reply:
This is what the `setEnvironmentVariable` is in the library: https://github.com/vlucas/phpdotenv/blob/828d19e597052ddeee65890bb2b1a0912d79fea8/src/Loader.php#L353-L375

It's super close to what we have just we also have an extra layer of the local static `$env` too.

Reply:
> just we also have an extra layer of the local static $env too.

That is the critical point of this PR. That's not a cache, that's the key feature that solves the process isolation issue.